src/pages/docs/ai-transport/features/token-streaming/message-per-response.mdx

```diff
@@ -0,0 +1,373 @@
+   action to the Ably channel with an empty or the first token as content.
+2. **Token streaming**: Append subsequent tokens to the original message by publishing those tokens
+   with the `message.append` action.
+3. **Live Delivery**: Clients subscribed to the channel receive each appended token in realtime, allowing
+   them to progressively render the response.
+4. **Compacted history**: The channel history contains only one message per AI response,
+   which includes all tokens appended to it concatenated together.
+
+You do not need to mark the message or token stream as completed; the final message will
+automatically have the full response with all tokens appended to it.
+
+## Publishing tokens
+
+Publish tokens from a [Realtime](/docs/api/realtime-sdk) client, which maintains a
+persistent connection to the Ably service. This allows you to publish at very high message rates
+with the lowest possible latencies, while preserving guarantees around message delivery order.
+For more information, see [Realtime and REST](/docs/basics#realtime-and-rest).
+
+[Channels](/docs/channels) separate message traffic into different topics.
+For token streaming, each conversation or session typically has its own channel.
+
+Use the [`get()`](/docs/api/realtime-sdk/channels#get) method to create or retrieve a channel instance:
+
+<Code>
+```javascript
+const channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}');
+```
+</Code>
+
+
+To start streaming an AI response, publish the initial message. Then append each subsequent token
+to that message as it arrives from the AI model:
+<Code>
+```javascript
+// set the initial data to empty string or the first token
+// capture the msgSerial for appending subsequent tokens
+
+const { serials: [msgSerial] } = await channel.publish('ai-response', { data: '' });
+
+for await (const token of streamedResponse) {
+  // Append each token as it arrives
+  channel.appendMessage(msgSerial, token);
+}
+```
+</Code>
+
+When publishing tokens, don't await the `channel.appendMessage()` call. Ably rolls up acknowledgments
+and debounces them for efficiency, which means awaiting each append would unnecessarily slow down
+your token stream. Messages are still published in the order that `appendMessage()` is called, so delivery
+order is not affected.
+
+Append only supports concatenating data of the same type as the original message. For example, if
+the initial message data is a string, all appended tokens must also be strings. If the initial
+message data is binary, all appended tokens must be binary.
+
+This pattern allows publishing append operations for multiple concurrent model responses on the same
+channel. As long as you append to the correct message serial, tokens from different responses will
+not interfere with each other. And the final concatenated message for each response will contain only the tokens
+from that response.
+
+### Complete publish example
```

Comment:
After this header we should have some preamble text describing what the example shows